crates/egui/src/atomics/atomic_layout.rs

```diff
@@ -0,0 +1,392 @@
+/// [`AllocatedAtomicLayout`] for interaction styling.
+pub struct AtomicLayout<'a> {
+    id: Option<Id>,
+    pub atomics: Atomics<'a>,
+    gap: Option<f32>,
+    pub(crate) frame: Frame,
+    pub(crate) sense: Sense,
+    fallback_text_color: Option<Color32>,
+    min_size: Vec2,
+    wrap_mode: Option<TextWrapMode>,
+    align2: Option<Align2>,
+}
+
+impl Default for AtomicLayout<'_> {
+    fn default() -> Self {
+        Self::new(())
+    }
+}
+
+impl<'a> AtomicLayout<'a> {
+    pub fn new(atomics: impl IntoAtomics<'a>) -> Self {
+        Self {
+            id: None,
+            atomics: atomics.into_atomics(),
+            gap: None,
+            frame: Frame::default(),
+            sense: Sense::hover(),
+            fallback_text_color: None,
+            min_size: Vec2::ZERO,
+            wrap_mode: None,
+            align2: None,
+        }
+    }
+
+    /// Insert a new [`Atomic`] at the end of the list (right side).
+    #[inline]
+    pub fn push(mut self, atomic: impl Into<Atomic<'a>>) -> Self {
+        self.atomics.push(atomic.into());
+        self
+    }
+
+    /// Insert a new [`Atomic`] at the beginning of the list (left side).
+    #[inline]
+    pub fn push_front(mut self, atomic: impl Into<Atomic<'a>>) -> Self {
+        self.atomics.push_front(atomic.into());
+        self
+    }
+
+    /// Set the gap between atomics.
+    ///
+    /// Default: `Spacing::icon_spacing`
+    #[inline]
+    pub fn gap(mut self, gap: f32) -> Self {
+        self.gap = Some(gap);
+        self
+    }
+
+    /// Set the [`Frame`].
+    #[inline]
+    pub fn frame(mut self, frame: Frame) -> Self {
+        self.frame = frame;
+        self
+    }
+
+    /// Set the [`Sense`] used when allocating the [`Response`].
+    #[inline]
+    pub fn sense(mut self, sense: Sense) -> Self {
+        self.sense = sense;
+        self
+    }
+
+    /// Set the fallback (default) text color.
+    ///
+    /// Default: [`crate::Visuals::text_color`]
+    #[inline]
+    pub fn fallback_text_color(mut self, color: Color32) -> Self {
+        self.fallback_text_color = Some(color);
+        self
+    }
+
+    /// Set the minimum size of the Widget.
+    #[inline]
+    pub fn min_size(mut self, size: Vec2) -> Self {
+        self.min_size = size;
+        self
+    }
+
+    /// Set the [`Id`] used to allocate a [`Response`].
+    #[inline]
+    pub fn id(mut self, id: Id) -> Self {
+        self.id = Some(id);
+        self
+    }
+
+    /// Set the [`TextWrapMode`] for the [`Atomic`] marked as `shrink`.
+    ///
+    /// Only a single [`Atomic`] may shrink. If this (or `ui.wrap_mode()`) is not
+    /// [`TextWrapMode::Extend`] and no item is set to shrink, the first (right-most)
+    /// [`AtomicKind::Text`] will be set to shrink.
+    #[inline]
+    pub fn wrap_mode(mut self, wrap_mode: TextWrapMode) -> Self {
+        self.wrap_mode = Some(wrap_mode);
+        self
+    }
+
+    /// Set the [`Align2`].
+    ///
```

Comment:
What is being aligned? Is it the atomics within the AtomicLayout, or the whole AtomicLayout within the parent?